packages/core/shared/types.ts

```diff
@@ -262,3 +264,17 @@ export type AssistantMessage = {
     };
   }>;
 };
+
+/*
+ * A data message is an application-specific message from the assistant
+ * that should be shown in order with the other messages.
+ *
+ * It can trigger other operations on the frontend, such as annotating
+ * a map.
+ */
+export type DataMessage = {
+  id?: string; // optional id, implement if needed (e.g. for persistance)
+  role: 'data';
+  text?: string; // optional text to display in the UI
+  data: JSONValue; // application-specific data
```

Comment:
Given the use case and semantics, I'd prefer the name `description?: string` over `text?: string`.